pkgs/by-name/le/lemminx/package.nix

```diff
@@ -2,11 +2,22 @@
 , fetchFromGitHub
 , makeWrapper
 , jre_headless
+, jre_minimal
 , maven
 , writeScript
 , lemminx
 }:
 
+let
+  jre = jre_minimal.override {
+    modules = [
+      "java.base"
+      "java.logging"
+      "java.xml"
+    ];
+    jdk = jre_headless;
```

Comment:
```suggestion
```
This looks odd. What is the intention behind setting a jdk to jre? Isn't this only a buildInput and we could just use the default jdk and thus omit this?

Reply:
I actually first tried that and it used the none headless jdk. Also since java 9 (iirc) jre and jdk are the same.

Reply:
@SuperSandro2000 I see. Does the closure size also differ whether you use `jdk = jre_headless` or if you omit it completely? It is only a buildInput at the moment so it should not matter. If it does not matter then I would just omit it.

Reply:
> It is only a buildInput at the moment so it should not matter.

It is used in the wrapper line 81 which I also changed.

And yeah, it matters.
With the default we get:
```
 ➜ nix path-info -Sh ./result
/nix/store/29x2krdp8mp4naibigv3xg2vniainmsl-lemminx-0.27.0       453.2 MiB
```
with headless only:
```
 ➜ nix path-info -Sh ./result
/nix/store/jk5l41918vdhjklqc6nwmlg5xdf68rph-lemminx-0.27.0       113.6 MiB
```


Also see https://github.com/NixOS/nixpkgs/blob/7edc243443b44444eba596557de03ee52beca2eb/pkgs/top-level/all-packages.nix#L15911-L15919

Reply:
@SuperSandro2000 I was never talking about the jre you are using in the wrapper script. I was always talking about the confusing fact that you built a custom jre with another jre. Usually you build a custom jre with a jdk. I opened a new pull request here to clean this up: https://github.com/NixOS/nixpkgs/pull/327915.

Please don't speed merge your own pull requests without resolving the conversation with the maintainer first in the future.

Reply:
> I was always talking about the confusing fact that you built a custom jre with another jre.

There is how the jre_minimal function works. see https://github.com/NixOS/nixpkgs/blob/60de3306b0170a11da90e23207baa2eda27b7fc3/doc/languages-frameworks/java.section.md#:~:text=modules%20parameter%20on-,jre_minimal,-to%20build%20a


> Usually you build a custom jre with a jdk.

We are not building a custom jre/jdk. We just collect the modules with jmods here https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/openjdk/jre.nix#L24

And jre/jdk since java 9 is the same. See the comment at https://github.com/NixOS/nixpkgs/blob/7edc243443b44444eba596557de03ee52beca2eb/pkgs/top-level/all-packages.nix#L15911-L15919

And as we are just clearing up questions at this point and not even discuss the code changes.